src/notation/view/notationnavigator.cpp

```diff
@@ -182,11 +212,17 @@ void NotationNavigator::setCursorRect(const QRectF& rect)
 
     RectF newCursorRect = notationContentRect().intersected(RectF::fromQRectF(rect));
 
-    moveCanvasToRect(newCursorRect);
-
+    bool moved = moveCanvasToRect(newCursorRect);
     m_cursorRect = newCursorRect;
+
+    m_viewRect->setSize(this->size());
+    m_viewRect->setRect(fromLogical(newCursorRect));
+
     rescale();
```

Comment:
Shouldn't the cursor also be updated when the Navigator is rescaled?

Reply:
Yeah, good point. I moved these lines after the `rescale()` call.